qemu/tests/ovs_host_vlan.py

```diff
@@ -0,0 +1,249 @@
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+    txt = "Setup netperf in guest."
+    image_snapshot = params.get("image_snapshot")
+    error.context(txt, logging.info)
+    netperf_setup(test, params, env)
+    params["image_snapshot"] = image_snapshot
+
+    os_type = params.get("os_type", "linux")
+    bridge_name = params.get("private_vlan_bridge", "ovs_private_br")
+    login_timeout = int(params.get("login_timeout", 360))
+
+    txt = "Stop NetworkManager service in host."
+    error.context(txt, logging.info)
+    cmd = "service NetworkManager stop"
+    utils.system(cmd, timeout=120, ignore_status=True)
+
+    txt = "Create a new private ovs bridge, which has"
+    txt += " no physical nics inside."
+    error.context(txt, logging.info)
+    create_ovs_cmd = "ovs-vsctl add-br %s" % bridge_name
+    try:
+        utils.system(create_ovs_cmd, timeout=120, ignore_status=False)
+    except error.CmdError:
+        raise error.TestFail("Fail to create ovs bridge %s" % bridge_name)
+
+    try:
+        params["start_vm"] = "yes"
+        params["netdst"] = bridge_name
+        vms = params.get("vms").split()
+        sessions = []
+        ips = []
+        txt = "Start multi vms and add them to 2 vlans."
+        error.context(txt, logging.info)
+        for vm_name in vms:
+            vm_params = params.object_params(vm_name)
+            env_process.preprocess_vm(test, vm_params, env, vm_name)
+        for vm_name in vms:
+            vm_params = params.object_params(vm_name)
+            change_ip_cmd = vm_params.get("change_ip_cmd", "ifconfig %s %s")
+            vm = env.get_vm(vm_name)
+            ifname = vm.virtnet[0]["ifname"]
+            guest_ip = vm.virtnet[0].ip
+            vlan = vm_params.get("ovs_port_vlan")
+            create_port_cmd = "ovs-vsctl set Port %s tag=%s" % (ifname, vlan)
+            try:
+                output = ""
+                output = utils.system_output(create_port_cmd, timeout=120,
+                                             ignore_status=False)
+            except error.CmdError:
+                err = "Fail to create ovs port %s on bridge %s." % (ifname,
+                                                                    bridge_name)
+                err += " Command: %s, output: %s." % (create_port_cmd, output)
+                raise error.TestFail(err)
+
+            session_ctl = vm.wait_for_serial_login(timeout=login_timeout)
+            txt = "Stop NetworkManager service in guest %s." % vm_name
+            session_ctl.cmd("service NetworkManager stop",
+                            ignore_all_errors=True)
+
+            txt = "Set guest %s IP to %s" % (vm_name, guest_ip)
+            error.context(txt, logging.info)
+            mac = vm.get_mac_address()
+            utils_net.set_guest_ip_addr(session_ctl, mac, guest_ip,
```

Comment:
Tested result:
    File "/root/autotest/client/tests/virt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/ovs_host_vlan.py", line 163, in run
      utils_net.set_guest_ip_addr(session_ctl, mac, guest_ip,
  AttributeError: 'module' object has no attribute 'set_guest_ip_addr'